framework/src/eventloop/eventloop_event.c

```diff
@@ -81,283 +78,321 @@ static event_group_t *get_event_group(int type)
 	return ptr;
 }
 
-static bool is_registered_event_cb(el_event_t *handle)
+static void eventloop_unregister_event_type(int type)
 {
-	event_group_t *group_ptr;
-	event_node_t *node_ptr;
+	el_handle_t *handle;
+	event_group_t *event_group;
 
-	if (handle == NULL) {
-		return false;
+	if (type < 0 || type >= EL_EVENT_MAX) {
+		return;
 	}
 
-	group_ptr = (event_group_t *)sq_peek(&g_event_list);
-	while (group_ptr != NULL) {
-		node_ptr = (event_node_t *)sq_peek(&group_ptr->event_list);
-		while (node_ptr != NULL && node_ptr->handle != NULL) {
-			if (node_ptr->handle == handle) {
-				return true;
+	event_group = get_event_group(type);
+	if (event_group != NULL) {
+		handle = (el_handle_t *)sq_peek(&event_group->event_list);
+		while (handle) {
+			if (handle->pid == getpid()) {
+				sq_rem((FAR sq_entry_t *)handle, &event_group->event_list);
+				eventloop_decrease_app_handle_cnt(handle->pid);
+				EL_FREE(handle);
+				break;
 			}
-			node_ptr = (event_node_t *)sq_next(node_ptr);
+			handle = (el_handle_t *)sq_next(handle);
+		}
+		if (sq_empty(&event_group->event_list)) {

```

Comment:
we don't have to call sq_init with both of event_group & handle to clean?

Reply:
Done